pandas/core/arrays/arrow/array.py

```diff
@@ -825,7 +829,17 @@ def isna(self) -> npt.NDArray[np.bool_]:
 
         return self._pa_array.is_null().to_numpy()
 
-    def any(self, *, skipna: bool = True, **kwargs):
+    # error: Signature of "any" incompatible with supertype
+    # "ExtensionArraySupportsAnyAll"
+    @overload  # type: ignore[override]
+    def any(self, *, skipna: Literal[True] = ..., **kwargs) -> bool:
+        ...
+
+    @overload
+    def any(self, *, skipna: bool, **kwargs) -> bool | NAType:
+        ...
+
+    def any(self, *, skipna: bool = True, **kwargs) -> bool | NAType:
```

Comment:
Is this deviation from the base class on purpose (base class returns always a bool)?

Reply:
I think so. This can return NA with `skipna=False`

```python
In [5]: pd.Series([pd.NA], dtype="bool[pyarrow]").any(skipna=False)
Out[5]: <NA>

In [6]: pd.Series([pd.NA], dtype="boolean").any(skipna=False)
Out[6]: <NA>
```

Reply:
Okay - I will adjust the annotations in the base class to reflect that